### PR TITLE
chore(tests): make test output assertions simpler

### PR DESCRIPTION
**Pull request by mehulkar**
less output to match and keep track of when things change

### Testing Instructions

Closes TURBO-1609

### CODE REVIEW

File: turborepo-tests/integration/tests/find_turbo/hoisted.t [hunk 1]
```diff
@@ -4,24 +4,22 @@ Setup
 
 Make sure we use local and do not pass --skip-infer to old binary
   $ ${TESTDIR}/set_version.sh $(pwd) "1.0.0"
-  $ ${TURBO} build --filter foo -vv
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Global turbo version: .* (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Repository Root: .*/hoisted.t (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo path: .*/hoisted.t/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo version: 1.0.0 (re)
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/hoisted.t:11]: I think this is actually the line we want to check and the rest we can ignore aside from the last line where we verify that `--skip-infer` wasn't passed.

File: turborepo-tests/integration/tests/find_turbo/hoisted.t [hunk 2]
```diff
@@ -4,24 +4,22 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/hoisted.t/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package false (re)
+
+  $ ${TURBO} build --filter foo -vv > out.log
+  $ cat out.log | grep "Repository Root"
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/hoisted.t:9]:
```suggestion
  $ cat out.log | grep -o -F "Local turbo version: 1.0.0"
```

File: turborepo-tests/integration/tests/find_turbo/hoisted.t [hunk 3]
```diff
@@ -4,24 +4,22 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Global turbo version: .* (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Repository Root: .*/hoisted.t (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo path: .*/hoisted.t/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo version: 1.8.0 (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/hoisted.t/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package true (re)
+
+  $ ${TURBO} build --filter foo -vv > out.log
+  $ cat out.log | grep "Repository Root"
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/hoisted.t:20]: Same as above
```suggestion
  $ cat out.log | grep -o -F "Local turbo version: 1.8.0"
```

File: turborepo-tests/integration/tests/find_turbo/linked.t [hunk 4]
```diff
@@ -2,30 +2,36 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/linked.t/node_modules/.pnpm/turbo-(darwin|linux|windows)-(64|arm64)@1.0.0/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package false (re)
+
+# Save output to file so we can look for specific things
+  $ ${TURBO} build --filter foo -vv > out.log
+
+  $ cat out.log | grep "Repository Root" # Repo root is correct
+  .* .*/linked.t (re)
+
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/linked.t:13]: I think we should keep the check of the local turbo version
```suggestion

  $ cat out.log | grep -o -F "Local turbo version: 1.0.0"
```

File: turborepo-tests/integration/tests/find_turbo/linked.t [hunk 5]
```diff
@@ -2,30 +2,36 @@ Setup
...
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package true (re)
+
+# Save output to file so we can look for specific things
+  $ ${TURBO} build --filter foo -vv > out.log
+
+# Repo root is correct
+  $ cat out.log | grep "Repository Root"
+  .* .*/linked.t (re)
+
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/linked.t:30]: Same as above
```suggestion

  $ cat out.log | grep -o -F "Local turbo version: 1.8.0"
```

**Review by chris-olszewski — COMMENTED** [review]
I think we could make use of `grep` a little more to make the snapshots a little more contained. We can avoid the leading `.*` in the snapshot regex for the timestamp by using `grep -o -e "Repository root: .*"` and grep will only return the line starting with `Repository root`

File: turborepo-tests/integration/tests/find_turbo/hoisted.t (hunk 1 above)
Comment (mehulkar) [reply turborepo-tests/integration/tests/find_turbo/hoisted.t:11]: Don't we care about the path that's used also?

File: turborepo-tests/integration/tests/find_turbo/unplugged_env_moved.t [hunk 6]
```diff
@@ -5,29 +5,17 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: No local turbo binary found at: .*/unplugged_env_moved.t/node_modules/turbo/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo path: .*/unplugged_env_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo version: 1.8.0 (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/unplugged_env_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package true (re)
+  $ ${TURBO} build --filter foo -vv > out.log
+  $ grep --quiet -oF "Local turbo version: 1.8.0" out.log
+  $ grep --quiet -E ".*Running local turbo binary in .*\/unplugged_env_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/unplugged_env_moved.t:19]:
```suggestion
  $ grep --quiet -F "Local turbo version: 1.8.0" out.log
  $ grep --quiet -E "Running local turbo binary in .*\/unplugged_env_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```

File: turborepo-tests/integration/tests/find_turbo/unplugged_moved.t [hunk 7]
```diff
@@ -4,28 +4,16 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: No local turbo binary found at: .*/unplugged_moved.t/node_modules/turbo/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo path: .*/unplugged_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo version: 1.0.0 (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/unplugged_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package false (re)
+  $ ${TURBO} build --filter foo -vv > out.log
+  $ grep --quiet -oF "Local turbo version: 1.0.0" out.log
+  $ grep --quiet -E ".*Running local turbo binary in .*\/unplugged_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/unplugged_moved.t:9]:
```suggestion
  $ grep --quiet -F "Local turbo version: 1.0.0" out.log
  $ grep --quiet -E "Running local turbo binary in .*\/unplugged_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```

File: turborepo-tests/integration/tests/find_turbo/unplugged_moved.t [hunk 8]
```diff
@@ -4,28 +4,16 @@ Setup
...
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: No local turbo binary found at: .*/unplugged_moved.t/node_modules/turbo/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo path: .*/unplugged_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Local turbo version: 1.8.0 (re)
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: Running local turbo binary in .*/unplugged_moved.t/.moved/unplugged/turbo-(darwin|linux|windows)-(64|arm64)-npm-1.0.0-520925a700/node_modules/turbo-(darwin|linux|windows)-(64|arm64)/bin/turbo (re)
-  
-  [-0-9:.TWZ+]+ \[DEBUG] turborepo_lib::shim: supports_skip_infer_and_single_package true (re)
+  $ ${TURBO} build --filter foo -vv > out.log
+  $ grep --quiet -oF "Local turbo version: 1.8.0" out.log
+  $ grep --quiet -E ".*Running local turbo binary in .*\/unplugged_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```
Comment (chris-olszewski) [inline turborepo-tests/integration/tests/find_turbo/unplugged_moved.t:17]:
```suggestion
  $ grep --quiet -F "Local turbo version: 1.8.0" out.log
  $ grep --quiet -E "Running local turbo binary in .*\/unplugged_moved\.t\/\.moved\/unplugged\/.*\/bin\/turbo" out.log
```

**Review by chris-olszewski — APPROVED** [review]
This is such an improvement! Just suggestion 2 changes that aren't consequential, I think they just simplify things:
 - Dropping leading `.*` from grep patterns as grep doesn't need the pattern to match the entire line, just a part
 - When we use `--quiet` we don't need `-o` since we're not omitting the output at all

**Pull request merged**